entname/entname.go

```diff
@@ -88,6 +118,57 @@ func AggregateMemberInterface(dp *DeviceParams, index int) (string, error) {
 	return namer.AggregateMemberInterface(index)
 }
 
+// Port returns the vendor-specific name of the physical interface with the
+// given port parameters.
+func Port(dp *DeviceParams, pp *PortParams) (string, error) {
+	namer, err := lookupNamer(dp)
+	if err != nil {
+		return "", err
+	}
+	npp, err := namerPortParams(pp, namer.IsFixedFormFactor())
+	if err != nil {
+		return "", err
+	}
+	return namer.Port(npp)
+}
+
+func namerPortParams(pp *PortParams, fixedFormFactor bool) (*namer.PortParams, error) {
+	if pp.SlotIndex < 0 {
+		return nil, fmt.Errorf("slot index cannot be negative: %d", pp.SlotIndex)
+	}
+	if pp.PICIndex < 0 {
+		return nil, fmt.Errorf("pic index cannot be negative: %d", pp.PICIndex)
+	}
+	if pp.PortIndex < 0 {
+		return nil, fmt.Errorf("port index cannot be negative: %d", pp.PortIndex)
+	}
+	if pp.ChannelIndex < 0 {
+		return nil, fmt.Errorf("channel index cannot be negative: %d", pp.ChannelIndex)
+	}
+	if pp.SlotIndex > 0 && fixedFormFactor {
+		return nil, fmt.Errorf("cannot have a non-zero slot index on a fixed form factor device")
+	}
```

Comment:
I wonder if a bunch of switch statements, each on an index, is more extensible, if you think there are modifications that will come to this logic. If this will not change much I think it's fine.